### PR TITLE
http_async_client: exit mod_init if tm is not loaded

### PR DESCRIPTION
**Pull request by grumvalski**
#### Pre-Submission Checklist

- [X] Commit message has the format required by CONTRIBUTING guide
- [X] Commits are split per component (core, individual modules, libs, utils, ...)
- [X] Each component has a single commit (if not, squash them into one commit)
- [X] No commits to README files for modules (changes must be done to docbook files
in `doc/` subfolder, the README file is autogenerated)

#### Type Of Change
- [X] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)

#### Checklist:

- [X] PR should be backported to stable branches
- [X] Tested changes locally
- [ ] Related to issue #XXXX (replace XXXX with an open issue number)

### CODE REVIEW

**Comment by henningw** [comment]
Thanks for the PR, merged

File: src/modules/http_async_client/http_async_client_mod.c [hunk 1]
```diff
@@ -295,7 +295,7 @@ static int mod_init(void)
 
 	if(load_tm_api(&tmb) < 0) {
 		LM_INFO("cannot load the TM-functions - async relay disabled\n");
-		memset(&tmb, 0, sizeof(tm_api_t));
+		return -1;
```
Comment (dilyanpalauzov) [inline src/modules/http_async_client/http_async_client_mod.c:298]: I think the line above LM_ERROR is bettet than LM_INFO.

**Review by dilyanpalauzov — COMMENTED** [review]
I think the line above LM_ERROR is bettet than LM_INFO.

**Comment by henningw** [comment]
@dilyanpalauzov You are right, done in the commit above.

**Pull request merged**